src/Cli/dotnet/Commands/Test/MTP/MSBuildUtility.cs

```diff
@@ -18,28 +20,55 @@ internal static class MSBuildUtility
 {
     private const string dotnetTestVerb = "dotnet-test";
 
+    // https://github.com/dotnet/msbuild/pull/7992 :/
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "ProjectShouldBuild")]
+    static extern bool ProjectShouldBuild(SolutionFile solutionFile, string projectFile);
+
     public static (IEnumerable<ParallelizableTestModuleGroupWithSequentialInnerModules> Projects, bool IsBuiltOrRestored) GetProjectsFromSolution(string solutionFilePath, BuildOptions buildOptions)
     {
-        SolutionModel solutionModel = SlnFileFactory.CreateFromFileOrDirectory(solutionFilePath, includeSolutionFilterFiles: true, includeSolutionXmlFiles: true);
-
         bool isBuiltOrRestored = BuildOrRestoreProjectOrSolution(solutionFilePath, buildOptions);
 
         if (!isBuiltOrRestored)
         {
             return (Array.Empty<ParallelizableTestModuleGroupWithSequentialInnerModules>(), isBuiltOrRestored);
         }
 
-        string rootDirectory = solutionFilePath.HasExtension(".slnf") ?
-                Path.GetDirectoryName(solutionModel.Description)! :
-                SolutionAndProjectUtility.GetRootDirectory(solutionFilePath);
+        var msbuildArgs = MSBuildArgs.AnalyzeMSBuildArguments(buildOptions.MSBuildArgs, CommonOptions.PropertiesOption, CommonOptions.RestorePropertiesOption, CommonOptions.MSBuildTargetOption(), CommonOptions.VerbosityOption());
+        var solutionFile = SolutionFile.Parse(solutionFilePath);
+        var globalProperties = CommonRunHelpers.GetGlobalPropertiesFromArgs(msbuildArgs);
 
-        FacadeLogger? logger = LoggerUtility.DetermineBinlogger([.. buildOptions.MSBuildArgs], dotnetTestVerb);
+        globalProperties.TryGetValue("Configuration", out var activeSolutionConfiguration);
+        globalProperties.TryGetValue("Platform", out var activeSolutionPlatform);
 
-        var msbuildArgs = MSBuildArgs.AnalyzeMSBuildArguments(buildOptions.MSBuildArgs, CommonOptions.PropertiesOption, CommonOptions.RestorePropertiesOption, CommonOptions.MSBuildTargetOption(), CommonOptions.VerbosityOption());
+        if (string.IsNullOrEmpty(activeSolutionConfiguration))
+        {
+            activeSolutionConfiguration = solutionFile.GetDefaultConfigurationName();
+        }
+
+        if (string.IsNullOrEmpty(activeSolutionPlatform))
+        {
+            activeSolutionPlatform = solutionFile.GetDefaultPlatformName();
+        }
+
+        var solutionConfiguration = solutionFile.SolutionConfigurations.FirstOrDefault(c => c.ConfigurationName == activeSolutionConfiguration && c.PlatformName == activeSolutionPlatform)
+            ?? throw new InvalidOperationException($"The solution configuration '{activeSolutionConfiguration}|{activeSolutionPlatform}' is invalid.");
+
+        // TODO: What to do if the given key doesn't exist in the ProjectConfigurations?
```

Comment:
@rainersigwald @baronfel Few lines below, I'm filtering projects only where the project configurations contains the requested solution configuration. Should that be an error instead if the solution configuration isn't found in the dictionary? or is it a valid scenario? And if valid scenario, is skipping the project the right behavior?